### PR TITLE
Fix GoneDocumentSubscriber by passing sourceHost to RedirectRouteManager

### LINKED ISSUE #71: Deleting a Page fails if "Gone on remove" is enabled
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | -
| SuluRedirect Version | 2.0.0
| Browser Version | -

#### Actual Behavior

If gone on remove is enabled and you try to remove a page in admin it fails. 

GoneDocumentSubscriber try to save the page https://github.com/sulu/SuluRedirectBundle/blob/2.x/GoneSubscriber/GoneDocumentSubscriber.php#L100

but saveBydata method fails due to missing 'sourceHost' key

https://github.com/sulu/SuluRedirectBundle/blob/2.x/Manager/RedirectRouteManager.php#L41

#### Expected Behavior

The page is removed and redirect is created .

#### Possible Solutions

- Add coalescence operator in 'sourceHost' 
- Refactor method to set all arguments with typehints (BC break)

### PR DESCRIPTION
**Pull request by niklasnatter**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #71
| License | MIT

#### Why?

See #71

### CODE REVIEW

File: GoneSubscriber/GoneDocumentSubscriber.php [hunk 1]
```diff
@@ -97,7 +97,13 @@ public function createRedirects(RemoveEvent $event): void
 
         foreach ($this->getUrls($document) as $url) {
             try {
-                $this->redirectRouteManager->saveByData(['source' => $url, 'statusCode' => 410, 'enabled' => true, 'target' => '']);
+                $this->redirectRouteManager->saveByData([
+                    'source' => $url,
+                    'sourceHost' => null,
```
Comment (alexander-schranz) [inline GoneSubscriber/GoneDocumentSubscriber.php:102]: Here we need to set the webspace domain for `pages` aslong as there is no `{host}` replacer set. Else if I have a page `/test` in 2 webspaces and remove both it should redirect to the correct domain.

Comment (niklasnatter) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: How could I get the correct domain? I think there could be multiple configured domains for a single webspace, right?

Comment (niklasnatter) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: But not sure if we need to be strict here. I think the worst thing that can happen is that we return a `410` in a webspace that never had a page on that URL, right?

Comment (alexander-schranz) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: 2 possibilities:

A. we create a redirect for every configured domain for the current environment
B. create a redirect only for the main / first domain found in the webspace config for the current environment

For me both is okay.

Comment (alexander-schranz) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: we already generate all `urls` here https://github.com/sulu/SuluRedirectBundle/blob/d5b52d4fbb2da0fba34b4d67bb0e481949ed8c85/GoneSubscriber/GoneDocumentSubscriber.php#L158 but parsing away the domain

Comment (alexander-schranz) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: > But not sure if we need to be strict here. I think the worst thing that can happen is that we return a 410 in a webspace that never had a page on that URL, right?

🤔 thats correct maybe we stay with this solution currently then.

Comment (niklasnatter) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: Thanks! But how can I find out if there was a `{host}` replacer in the URL? 🙈

Comment (alexander-schranz) [reply GoneSubscriber/GoneDocumentSubscriber.php:102]: Think thats not longer possible as we don't have a method for it 🤔

**Pull request merged**